client/build.rs

```diff
@@ -17,14 +17,14 @@ fn main() {
     // TODO: check if this was causing the recursive loop.
     // println!("cargo:rerun-if-changed=..");
```

Comment:
@appetrosyan does it even work right with this line commented? I think no.
For example if @mversic will change something in API for smartcontracts in #1973 then a test using relevant smartcontract will pass just because it will use a previously compiled smartcontract version

Reply:
> For example if @mversic will change something in API for smartcontracts in https://github.com/hyperledger/iroha/pull/1973 then a test using relevant smartcontract will pass just because it will use a previously compiled smartcontract version. 

Two considerations: 
1. Each of the dependencies of `iroha_client` will trigger a re-compilation. If Marin made changes to the smartcontracts API then yes, since `iroha_client` doesn't depend on it, it might pass. The solution is to make that crate a `dev-dependency`. I didn't do that because of 2.
2. The only reason why you wouldn't re-compile the smartcontract is if you had the binary locally. In the CI, the binaries are excluded from the cache, so we don't really have a problem. 

I would prefer, that we use the actual dependencies, though, since `rerun-if-changed` really is meant to track non-rust source code and POD files that affect code generation, not rust code, which is the only thing that should re-trigger the compilation.

Reply:
On a side note, the test that this `build.rs` was introduced for, doesn't have the issue of passing when it shouldn't. Quite the oppposite.

Reply:
> On a side note, the test that this build.rs was introduced for, doesn't have the issue of passing when it shouldn't. Quite the oppposite.

Oh, yeah, I forgot about it. I'm not sure where the problem is. Testing time-triggers is hard...

Reply:
Thanks, now it's clear for me. I'm closing this conversation